### PR TITLE
feat(api): support chat logit_bias

### PR DESCRIPTION
**Pull request by Thump604**
## Summary

Adds OpenAI-compatible `logit_bias` support for chat completions by converting
the request map into a normal request-local logits processor.

## Scope

- Adds `ChatCompletionRequest.logit_bias`.
- Coerces OpenAI token-id string keys to integer token ids before calling
  `mlx_lm.sample_utils.make_logits_processors(logit_bias=...)`.
- Appends the generated processor to existing request-local processors so it can
  coexist with structured-output constraints and other decode controls.
- Keeps this separate from the Gemma 4 TextModel dispatch fix in #595.

This is a partial fix for #590. It does not change Gemma attention/mask handling
or model dependency floors.

## Validation

- `.venv/bin/python -m pytest tests/test_api_models.py::TestChatCompletion::test_request_accepts_logit_bias -q`
- `.venv/bin/python -m pytest tests/test_server.py::TestPromptCanonicalization::test_prepare_chat_completion_forwards_logit_bias -q`
- `.venv/bin/python -m pytest tests/test_api_models.py tests/test_server.py -q`
- `.venv/bin/ruff check vllm_mlx/api/models.py vllm_mlx/server.py tests/test_api_models.py tests/test_server.py`
- `git diff --check`

### CODE REVIEW

**Review by janhilgard — APPROVED** [review]
Reviewed the diff end-to-end. Approving — the change is small, well-scoped, and the integration with the existing processor pipeline is correct.

**Verified:**

- `mlx_lm.sample_utils.make_logits_processors` signature on a current install confirms `logit_bias: Dict[int, float] | None` — the string→int coercion at the API boundary is the right move, and matches OpenAI's wire format where keys arrive as strings.
- Processor ordering in `_prepare_chat_completion_invocation` is now logit_bias → json (response_format) → thinking-wrapper, which is consistent with how each layer composes (thinking wraps json as inner delegate; logit_bias is independent and additive).
- The updated thinking-processor block at the tail correctly strips only the trailing `json_logits_processor` before appending `thinking_proc`, so the logit_bias processor is preserved when both response_format and thinking are active. The `is`-identity check is the right way to detect "this is the one thinking wraps."
- `_coerce_logit_bias` returns HTTP 400 with a descriptive message on bad token-id strings — good error surface for clients that send float-looking strings or non-numeric junk.
- Tests exercise both the request-model acceptance and the wiring through `_prepare_chat_completion_invocation`, including verifying that the keys are int-coerced when the processor factory is called. Local CI (lint, type-check, tests, apple-silicon × 2, matrix × 4) is green on `3800f59`.

**Non-blocking follow-ups (out of scope for this PR, mentioning for visibility):**

1. `CompletionRequest` doesn't grow a `logit_bias` field in this PR — the legacy `/v1/completions` route stays unaffected. Worth a follow-up if you want full OpenAI parity, but the chat path is by far the dominant one and the scope split is sensible.
2. OpenAI clamps `logit_bias` values to `[-100, 100]`. The PR forwards floats unchanged; in practice `mlx_lm` will happily apply larger magnitudes, which effectively force/forbid tokens. Probably fine for an internal API, but consider clamping or warning at the boundary if this is meant to be a drop-in OpenAI replacement.
3. There's no validation that integer token ids are in vocab. Out-of-range entries should be silently no-ops in `make_logits_processors`, but it's worth a quick sanity check against `len(tokenizer)` at request time if you want a clean 400 instead of a possibly confusing acceptance.

None of those should hold this up.
